client/landing/gutenboarding/components/plans-button/index.tsx

```diff
@@ -12,33 +12,62 @@ import { useI18n } from '@automattic/react-i18n';
  * Internal dependencies
  */
 import JetpackLogo from 'components/jetpack-logo'; // @TODO: extract to @automattic package
-import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
-import { freePlan, defaultPaidPlan, getPlanSlugByPath, getPlanTitle } from '../../lib/plans';
-import { usePlanRouteParam } from '../../path';
+import { STORE_KEY as PLANS_STORE } from '../../stores/plans';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
 const PlansButton: React.FunctionComponent< Button.ButtonProps > = ( { ...buttonProps } ) => {
+	const [ isOpen, setIsOpen ] = React.useState( false );
 	const { __ } = useI18n();
 
 	// mobile first to match SCSS media query https://github.com/Automattic/wp-calypso/pull/41471#discussion_r415678275
 	const isDesktop = useViewportMatch( 'mobile', '>=' );
-
-	const hasPaidDomain = useSelect( ( select ) => select( ONBOARD_STORE ).hasPaidDomain() );
-	const planPath = usePlanRouteParam();
-	const plan = getPlanSlugByPath( planPath ) || ( hasPaidDomain ? defaultPaidPlan : freePlan );
```

Comment:
By removing this, we are breaking 2 existing functionalities (detailed in #41471 and #41679):
- getting plan from URL.
- replacing Free plan (shown by default) with a paid default version (eg: Personal) when the user picks a paid domain.

I suggest creating a selector for [defaultPlan](https://github.com/Automattic/wp-calypso/pull/41857/files#diff-a533017ba250fed92642a7cb13d6a7c6R39) and calling `useEffect` with `planPath ` as dependency to dispatch `setPlan` only when there is no selected plan.